File: _old_web_projects/samiam/webserverPrograming/guestbook.php
<!--
# Assignment: 08. PHP Guestbook (10/17/02)

-->

<html>
<head>
<Title>PHP Guestbook</title>
</head>
<body>
<?php
    Print "<h3>PHP Guestbook</h3>
           <p>
   
   <!-- /* Create form */ -->
    <form action = 'guestbook.php' method = 'POST'>
        Name:<br>
        <input type = 'text' name = 'name'>
        <p></p>
        Message:
        <br>
        <textarea name='message' value='' rows='10' cols='50'></textarea>
        <p></p>
        <input type = 'Submit'>
    </form>

    <p></p>
    <hr align=left width=425>
    <p></p>";

    /* Connecting, selecting database */
    $link = mysql_connect("localhost", "n3loobko", "n3loobko")
        or die("Could not connect");
    mysql_select_db("n3loobko") or die("Could not select database");
  
    
    /* enter new message (if there is one) */
    if ($_POST["name"] && $_POST["message"]) {
      $name = $_POST["name"];
      $message = $_POST["message"];
      $query = "INSERT INTO guestbook (Name, Message) VALUES ('$name', '$message')";
      $result = mysql_query($query) or die("Query failed");
    }


    /* Performing SQL query */
    $query = "SELECT Message, Name, Date FROM guestbook ORDER BY Date DESC";
    $result = mysql_query($query) or die("Query failed");

    /* Printing results in HTML */ 
    while ($row = mysql_fetch_array($result)) {
    print "<table width =425>\n";
    print "\t<tr><td><b>";
    print $row["Message"];
    print "<br></b></td></tr>\n";
    print "\t<tr><td>";
    print $row["Name"];
    print "</td></tr>\n";
    if (ereg ("([0-9]{4})([0-9]{2})([0-9]{2})([0-9]{2})([0-9]{2})([0-9]{2})", $row["Date"], $regs)) {
      print "\t<tr><td>";     
      print "$regs[2].$regs[3].$regs[1] $regs[4]:$regs[5]:$regs[6]";
      print "</td></tr>\n";
    }
    print "</table>";
    Print "<p></p>";
    Print "<hr align=left width=425>";
    Print "<p></p>";
    }

    /* Free resultset */
    mysql_free_result($result);

    /* Closing connection */
    mysql_close($link);
?>

File: _old_web_projects/samiam/webserverPrograming/guestbook.pl
#!/usr/bin/perl -w

# Assignment: 06. PERL Guestbook (10/3/02)

use DBI;
use CGI;
use strict;
my $q = new CGI;

my $database = 'n3loobko';
my $user = 'n3loobko';
my $password = 'n3loobko';

# Create form
print $q->header;
print $q->start_html('Guestbook'),
      $q->start_form;
print $q->h3('Perl Guestbook');
print $q->p;
print 'Name:';
print $q->br();
print $q->textfield('name');
print $q->p;
print 'Message:';
print $q->br();
print $q->textarea('message','',10,50);
print $q->p;
print $q->submit(-value=>'submit message'),
      $q->end_form;
print '<hr align = left width = 425>';
print $q->p;

# open database
my $dbh = DBI->connect("DBI:mysql:$database", $user, $password);
my $sql = 'INSERT INTO guestbook (Name, Message) VALUES (?,?)';
my $sth = $dbh->prepare($sql);

# enter new message (if there is one)
my $name = $q->param('name');
my $message = $q->param('message');
$sth->execute($name, $message);

# get contents of database
$sql = 'SELECT * FROM guestbook ORDER BY Date DESC';
$sth = $dbh->prepare($sql);
my $rv = $sth->execute();

# Display messages
if($rv) {
  while(my @row = $sth->fetchrow_array) {
    if ($row[1] =~ /^(\d{4})(\d{2})(\d{2})(\d{2})(\d{2})(\d{2})$/) {  
      print qq|<b>$row[3]</b><p>$row[2]<br>$2/$3/$1 $4:$5:$6<p><hr align = left width = 425><p>|;
    }
  }
}

$q->end_html;

    
